packages/codegen.go/src/fake/servers.ts

```diff
@@ -193,13 +218,63 @@ export async function generateServers(codeModel: go.CodeModel): Promise<ServerCo
   return new ServerContent(operations, generateServerInternal(codeModel, requiredHelpers));
 }
 
-function generateServerTransportMethods(clientPkg: string, serverTransport: string, client: go.Client, finalMethods: Array<go.Method>, imports: ImportManager): string {
+// method names for fakes dispatching
+const dispatchMethodFake = 'dispatchToMethodFake';
+const dispatchToClientFake = 'dispatchToClientFake';
+
+function generateServerTransportDo(serverTransport: string, client: go.Client, finalSubClients: Array<go.Client>, finalMethods: Array<go.Method>): string {
   const receiverName = serverTransport[0].toLowerCase();
   let content = `// Do implements the policy.Transporter interface for ${serverTransport}.\n`;
   content += `func (${receiverName} *${serverTransport}) Do(req *http.Request) (*http.Response, error) {\n`;
   content += '\trawMethod := req.Context().Value(runtime.CtxAPINameKey{})\n';
   content += '\tmethod, ok := rawMethod.(string)\n';
   content += '\tif !ok {\n\t\treturn nil, nonRetriableError{errors.New("unable to dispatch request, missing value for CtxAPINameKey")}\n\t}\n\n';
+
+  if (finalSubClients.length > 0 && finalMethods.length > 0) {
+    // client contains client accessors and methods.
+    // if the method isn't for this client, dispatch to the correct client
+    content += `\tif client := method[:strings.Index(method, ".")]; client != "${client.name}" {\n`;
+    content += `\t\treturn ${receiverName}.${dispatchToClientFake}(req, client)\n\t}\n`;
+    // else dispatch to our method fakes
+    content += `\treturn ${receiverName}.${dispatchMethodFake}(req, method)\n`;
+  } else if (finalSubClients.length > 0) {
+    content += `\treturn ${receiverName}.${dispatchToClientFake}(req, method[:strings.Index(method, ".")])\n`;
+  } else {
+    content += `\treturn ${receiverName}.${dispatchMethodFake}(req, method)\n`;
+  }
+  content += '}\n\n'; // end Do
+  return content;
+}
+
+function generateServerTransportClientDispatch(serverTransport: string, subClients: Array<go.Client>, imports: ImportManager): string {
```

Comment:
This is pretty much the same thing we do in `src/fake/factory.ts`. It will likely get consolidated at some point.